middle_end/flambda2/to_cmm/to_cmm_primitive.ml

```diff
@@ -767,8 +770,10 @@ let unary_primitive env res dbg f arg =
         ~else_dbg:dbg )
   | Is_flat_float_array ->
     None, res, C.eq ~dbg (C.get_tag arg dbg) (C.floatarray_tag dbg)
-  | End_region | End_try_region ->
+  | End_region { ghost = false } | End_try_region { ghost = false } ->
     None, res, C.return_unit dbg (C.endregion ~dbg arg)
+  | End_region { ghost = true } | End_try_region { ghost = true } ->
+    None, res, C.unit ~dbg
```

Comment:
I'm not sure `unit` is the correct thing here. Considering that it is supposed to return a region, something along the lines of `1` would be better I'd think. That being said, we could also ignore/erase ghost regions at that point if that is the goal.